MCOL-4412

drrtuy opened the pull request:
The main idea is to replace literal empty magic values used in PrimProc and WriteEngineServer with pointers to constant values that are TypeHandler-specific so processing avoids type-based switch-case blocks.

drrtuy commented on datatypes/mcs_datatype.h:
```diff
@@ -2294,160 +2329,14 @@ class TypeHandlerSLongDouble: public TypeHandlerReal
     throw logging::QueryDataExcept("convertColumnData: unknown column data type.", logging::dataTypeErr);
     return boost::any();
   }
-};
```
The classes were moved to come after getEmptyTypeHandlerStr.

mariadb-AlexeyAntipovsky commented on writeengine/wrapper/writeengine.cpp:
```diff
@@ -775,9 +781,13 @@ int WriteEngineWrapper::deleteRow(const TxnID& txnid, const vector<CSCTypesList>
+            const uint8_t* emptyVal = m_colOp[op(curColStruct.fCompressionType)]->
+                getEmptyRowValue(curColStruct.colDataType, curColStruct.colWidth);
+
+            if (curColStruct.colWidth == datatypes::MAXDECIMALWIDTH)
```
Shouldn't we also check the column data type here?

drrtuy replied:
We shouldn't b/c we check coltype getting empty value ptr from BlockOp::getEmptyRowValue()

drrtuy committed:
MCOL-4412 Introduce TypeHandler::getEmptyValueForType to return const ptr for an empty value

WE changes for SQL DML and DDL operations

Changes for bulk operations

Changes for scanning operations

Cleanup
